src/legacy/ui/public/agg_types/controls/time_interval.tsx

```diff
@@ -0,0 +1,161 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { get, find } from 'lodash';
+import React, { useEffect } from 'react';
+
+import { EuiFormRow, EuiIconTip, EuiComboBox, EuiComboBoxOptionProps } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { AggParamEditorProps } from '../../vis/editors/default';
+import { AggParamOption } from '../agg_param';
+import { parseInterval } from '../utils';
+
+interface ComboBoxOption extends EuiComboBoxOptionProps {
+  key: string;
+}
+
+function TimeIntervalParamEditor({
+  agg,
+  aggParam,
+  editorConfig,
+  value = '',
+  setValue,
+  isInvalid,
+  setTouched,
+  setValidity,
+}: AggParamEditorProps<string>) {
+  const label = i18n.translate('common.ui.aggTypes.timeInterval.minimumIntervalLabel', {
+    defaultMessage: 'Minimum interval',
+  });
+  const interval = get(agg, 'buckets.getInterval') && agg.buckets.getInterval();
+  const isScaled = interval && interval.scaled;
+  const scaledHepText =
+    isScaled && !isInvalid ? (
+      <label>
+        <FormattedMessage
+          id="common.ui.aggTypes.timeInterval.scaledHelpText"
+          defaultMessage="Currently scaled to {bucketDescription}"
+          values={{ bucketDescription: get(interval, 'description') || '' }}
+        />{' '}
+        <EuiIconTip
+          position="right"
+          type="questionInCircle"
+          color="text"
+          content={interval.scale <= 1 ? tooManyBucketsTooltip : tooLargeBucketsTooltip}
+        />
+      </label>
+    ) : null;
+
+  const helpText = (
+    <>
+      {scaledHepText}
+      {get(editorConfig, 'interval.help') || selectOptionHelpText}
+    </>
+  );
+
+  const timeBase: string = get(editorConfig, 'interval.timeBase');
+  const options = (aggParam.options || []).reduce(
+    (filtered: ComboBoxOption[], option: AggParamOption) => {
+      if (option.enabled ? option.enabled(agg) : true) {
+        filtered.push({ label: option.display, key: option.val });
+      }
+      return filtered;
+    },
+    [] as ComboBoxOption[]
+  );
+  const definedOption = find(options, { key: value });
+  const selectedOptions = definedOption ? [definedOption] : [{ label: value, key: 'custom' }];
+  const errors = [];
+
+  if (isInvalid && value) {
+    errors.push(
+      i18n.translate('common.ui.aggTypes.timeInterval.invalidFormatErrorMessage', {
+        defaultMessage: 'Invalid interval format.',
+      })
+    );
+  }
+
+  const onCustomInterval = (customValue: string) => {
+    const normalizedCustomValue = customValue.trim();
+    const isValid = normalizedCustomValue ? parseInterval(normalizedCustomValue, timeBase) : false;
+
+    setValidity(isValid);
+    setValue(normalizedCustomValue);
+    setTouched();
```

Comment:
`EuiComboBox` invokes `onBlur` callback before `onChange` callbacks, so we set touched manually after validation.